caikit/runtime/model_management/loaded_model.py

```diff
@@ -118,7 +118,11 @@ def model(self) -> ModuleBase:
         self.wait()
         return self._model
 
-    def loaded(self) -> bool:
+    def loaded(self, require_instance: bool = False) -> bool:
+        # If require_instance is true then require that the loaded model has a constructed instance
+        # attached
+        if require_instance:
+            return bool(self._model)
         return bool(self._model or self._caikit_model_future.done())
```

Comment:
If we do keep this as an option:

```suggestion
        return (
            bool(self._model) or
            (not require_instance and self._caikit_model_future.done())
        )
```